### PR TITLE
op-node: persist p2p bans with expiration time, update ban-expiry gater

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

This generalizes the score-book implementation, so we can use the same pattern with peer-bans and IP-bans.

The new peerstore temp-ban interface is then used by the ban-expiry connection gater (not enabled in p2p node yet).
The connection gater now first checks the inner blocking gater (i.e. perma bans), and then completely independently checks temp-bans. This is important, since we want to prune the temp-ban data in the peerstore, but don't want it to cause inconsistencies with perma-bans, so I decoupled them. The ban-expiration record itself signifies a temporary ban.

This also adds `BanPeer(id, expiration)` and `BanIP(ip, expiration)` helpers to the p2p node. Unban by setting the time to 0 time struct. We should consider exposing these via the p2p RPC like we do with perma-bans.

**mocks are updated to match the new interface, line count is a bit high because of it**

**Tests**

- The score-book tests cover the generalized book, but the the IP and peer ban books don't have tests (yet).
- The connection gater tests were updated to match the new peer/IP ban store interface and independent behavior of inner-gater.

**Metadata**

Fix CLI-4011

### CODE REVIEW

File: op-node/p2p/gating/expiry.go [hunk 1]
```diff
@@ -47,19 +46,19 @@ func AddBanExpiry(gater BlockingConnectionGater, store ExpiryStore, log log.Logg
 
 func (g *ExpiryConnectionGater) peerBanExpiryCheck(p peer.ID) (allow bool) {
 	// if the peer is blocked, check if it's time to unblock
-	expiry, err := g.store.PeerBanExpiry(p)
+	expiry, err := g.store.GetPeerBanExpiration(p)
+	if err == store.UnknownBanErr {
```
Comment (tynes) [inline op-node/p2p/gating/expiry.go:50]: Is there a semantic different from doing an equality check with `==` and using `errors.Is`?

Comment (tynes) [reply op-node/p2p/gating/expiry.go:50]: I feel like `errors.Is` will catch wrapped errors where an equality check wont?

**Review by ajsutton — COMMENTED** [review]
Looks good.  I'd really like to see some tests around the new books go in. They're mostly leveraging the record book base so shouldn't have too many corner cases to try and cover.

File: op-node/p2p/gating/expiry.go (hunk 1 above)
Comment (protolambda) [reply op-node/p2p/gating/expiry.go:50]: `errors.Is` would work, but similar to `EOF` in many IO stdlib funcs, returning the exact error from `GetPeerBanExpiration` is part of the api here, so it shouldn't make a difference.

Comment (ajsutton) [reply op-node/p2p/gating/expiry.go:50]: True, I'm just paranoid. :)

**Comment by ajsutton** [comment]
Added the tests and made the `errors.Is` change to satisfy my paranoia. :)
